nuc_morph_analysis/analyses/density/figure_watershed_based_density_schematic.py

```diff
@@ -0,0 +1,229 @@
+            ax.set_ylim([y1,y1+h])
+
+
+            position = ax.get_position().bounds
+            if key == 'overlay':
+                draw_contours_on_image(ax,contour_list,filled=True,linewidth=0.5)
+            if key == 'colorize':
+                labels_in_img = np.unique(img_dict['pseudo_cells_img'][0][crop_exp])
+                dftsub = dft[dft['label_img'].isin(labels_in_img)]
+                colorfeat='2d_area_nuc_cell_ratio' #feature
+                cmapstr = 'cool' # colormap
+                color_values = dftsub[colorfeat].values
+                
+                draw_contours_on_image(ax,contour_list,filled=True,colorize=True,dft=dftsub,linewidth=0.5,colorfeat=colorfeat,cmapstr=cmapstr)
+
+                colorbarmin = np.nanmin(color_values)
+                colorbarmax = np.nanmax(color_values)
+                colorbar = plt.cm.ScalarMappable(cmap=cmapstr,norm=plt.Normalize(vmin=colorbarmin,vmax=colorbarmax))
+                colorbar.set_array([])
+                colorbar.set_clim(colorbarmin,colorbarmax)
+                cbar = plt.colorbar(colorbar,ax=ax,location='bottom')
+                _,label,unit,_ = get_plot_labels_for_metric(colorfeat)
+                cbar.set_label(f"{label} {unit}")
+                cbar.set_ticks([colorbarmin,colorbarmax])
+                cbar.set_ticklabels([f'{colorbarmin:.2f}',f'{colorbarmax:.2f}'])
+                
+            if key in ['distance']:
+
+                distance_scale = 0.108 # pixel size
+                if RESOLUTION_LEVEL == 1:
+                    distance_scale *= 2.5
+                colorbar = plt.cm.ScalarMappable(cmap=imgcmap,norm=plt.Normalize(vmin=vmin,vmax=vmax))
+                cbar = plt.colorbar(colorbar,ax=ax,location='bottom')
+                cbar.set_ticks([vmin,vmax])
+                cbar.set_ticklabels([f'{vmin*distance_scale:.2f}',f'{vmax*distance_scale:.2f}'])
+                cbar.set_label('Distance Transform (µm)')
+            
+            if key in ['distance','colorize']:
+                # adjust the axis back to its original position
+                ax.set_position(position)
+                # adjust position of colorbar
+                cbar_position = cbar.ax.get_position()
+                # cbar.ax.set_position([cbar_position.x0+0.05,cbar_position.y0,cbar_position.width,cbar_position.height]) # for left location
+                cbar.ax.set_position([cbar_position.x0,cbar_position.y0-0.15,cbar_position.width,cbar_position.height]) # for bottom location
+
+        # now save the figure
+        savedir = Path(__file__).parent / 'figures' / 'watershed_figure_illustration'
+        savedir.mkdir(exist_ok=True,parents=True)
+        
+        for ext in ['.png','.pdf']:
+            savename = f'{colony}_{track}_{TIMEPOINT}_{full_crop}_res{RESOLUTION_LEVEL}_{cmapstr}'
+            savepath = savedir / savename
+            print(f'Saved figure to {savepath}')
+            save_and_show_plot(str(savepath),
```

Comment:
The plots are getting cut off in the saved output. Maybe bbox_inches='tight' would fix it!